Update dartfn templates

subfuzion opened the pull request:
This updates the `dartfn` templates to use the latest example code for functions_framework 0.4.0 and correspondingly bumps the tool's version. The updated version will be published after the PR is merged.

grant reviewed (approved):
Looks OK to me. Please wait for @kevmoo to approve as well.

subfuzion commented:
> @kevmoo already approved, but update triggers a new approval requirement, @grant. If I addressed your nits appropriately, then go ahead and approve. Thx.
